Microsoft.Toolkit.Uwp.UI.Controls/Common/VisualStates.cs

```diff
@@ -0,0 +1,85 @@
+﻿// (c) Copyright Microsoft Corporation.
+// This source is subject to the Microsoft Public License (Ms-PL).
+// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
+// All other rights reserved.
+
+using System.Diagnostics;
+using System.Linq;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Names and helpers for visual states in the controls.
+    /// </summary>
+    internal static class VisualStates
+    {
+        /// <summary>
+        /// Use VisualStateManager to change the visual state of the control.
+        /// </summary>
+        /// <param name="control">
+        /// Control whose visual state is being changed.
+        /// </param>
+        /// <param name="useTransitions">
+        /// A value indicating whether to use transitions when updating the
+        /// visual state, or to snap directly to the new visual state.
+        /// </param>
+        /// <param name="stateNames">
+        /// Ordered list of state names and fallback states to transition into.
+        /// Only the first state to be found will be used.
+        /// </param>
+        public static void GoToState(Control control, bool useTransitions, params string[] stateNames)
+        {
+            Debug.Assert(control != null, "control should not be null!");
+            Debug.Assert(stateNames != null, "stateNames should not be null!");
+            Debug.Assert(stateNames.Length > 0, "stateNames should not be empty!");
+
+            foreach (string name in stateNames)
+            {
+                if (VisualStateManager.GoToState(control, name, useTransitions))
+                {
+                    break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the implementation root of the Control.
+        /// </summary>
+        /// <param name="dependencyObject">The DependencyObject.</param>
+        /// <remarks>
+        /// Implements Silverlight's corresponding internal property on Control.
```

Comment:
Yes, Silverlight has this property, but WinRT doesn't. So this method may be useful.